pkg/config/config.go

```diff
@@ -18,11 +19,30 @@ var (
 		"instance-url",
 		field.WithDescription(`The GitHub instance URL to connect to. (default "https://github.com")`),
 	)
+	appIDField = field.StringField(
+		"app-id",
+		field.WithDescription("The GitHub App to connect to."),
+	)
+	appPrivateKey = field.StringField(
+		"app-privatekey",
+		field.WithDescription("The private key used to connect to the GitHub App"),
+	)
 )
 
 //go:generate go run ./gen
 var Config = field.NewConfiguration([]field.SchemaField{
 	accessTokenField,
 	orgsField,
 	instanceUrlField,
+	appIDField,
+	appPrivateKey,
 })
+
+func ValidateConfig(cfg *Github) error {
+	apiKey := cfg.GetString(accessTokenField.FieldName)
+	appKey := cfg.GetString(appIDField.FieldName)
+	if len(apiKey) == 0 && len(appKey) == 0 {
+		return fmt.Errorf("api-key or app-privatekey is missing")
+	}
+	return nil
+}
```

Comment:
I think this can be defined using relationships.

https://github.com/ConductorOne/baton-databricks/blob/9cdba512da2e1b638b48721a588d900e13c63f83/pkg/config/schema.go#L62-L82